src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -461,8 +461,8 @@ def continuing_criterion(carry: OptimizationCarry) -> jtp.Bool:
                 terrain=model.terrain,
                 K=1e6,
                 D=2e3,
-                p=0.0,
-                q=0.0,
+                p=0.5,
+                q=0.5,
```

Comment:
This makes sense since we always use a non-linear model with $p$ and $q$ powers you updated. Considering that there's no tangential component ($\mu = 0$) the effect of using different powers is to get an initial guess with a lower magnitude. I don't think it will have a great effect since it's just the starting point, but it's worth investigating.

Reply:
The test was failing only because of that, so I'm wondering if there is something that can be improved on the optimization part, since it's very sensitive to the initialization

Reply:
Yes, this is also what I noticed during the last refactoring.